### PR TITLE
[SPARK-33547][SQL] Add usage of typed literal in doc

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
According  to https://github.com/apache/spark/pull/30421#discussion_r530024114
Add typed literal in doc

### Why are the changes needed?
Make user clear about usage of typed literal

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
NOT need

### CODE REVIEW

File: docs/sql-ref-literals.md [hunk 1]
```diff
@@ -21,14 +21,74 @@ license: |
 
 A literal (also known as a constant) represents a fixed data value. Spark SQL supports the following literals:
 
+ * [Typed Literal](#typed-literal)
  * [String Literal](#string-literal)
  * [Binary Literal](#binary-literal)
  * [Null Literal](#null-literal)
  * [Boolean Literal](#boolean-literal)
  * [Numeric Literal](#numeric-literal)
+ * [Timestamp Literal](#timestamp-literal)
```
Comment (AngersZhuuuu) [inline docs/sql-ref-literals.md:30]: Missed in current doc

**Comment by AngersZhuuuu** [comment]
FYI @maropu there is many duplicated example between typed literal and corresponding  literal, any suggestion?

**Review by srowen — COMMENTED** [review]
Seems OK to me.

File: docs/sql-ref-literals.md [hunk 2]
```diff
@@ -21,14 +21,74 @@ license: |
...
  * [Datetime Literal](#datetime-literal)
  * [Interval Literal](#interval-literal)
 
+### Typed Literal
```
Comment (maropu) [inline docs/sql-ref-literals.md:34]: I feel "Typed Literal" is a bit confusing. For example, a numerical literal is also a typed literal, right? How about saying `Typed Literal for Binary, Date, Timestamp, and Interaval` instead?

**Comment by maropu** [comment]
cc: @huaxingao

File: docs/sql-ref-literals.md (hunk 2 above)
Comment (AngersZhuuuu) [reply docs/sql-ref-literals.md:34]: > I feel "Typed Literal" is a bit confusing. For example, a numerical literal is also a typed literal, right? How about saying `Typed Literal for Binary, Date, Timestamp, and Interaval` instead?

LGTM, changed

File: docs/sql-ref-literals.md [hunk 3]
```diff
@@ -21,14 +21,74 @@ license: |
...
+[TYPE] '[VALUE]'
+```
+
+#### Parameters
+
+* **TYPE**
+   
+    `TYPE` parameter shows which type of Literal we will generate. Currently Date, Timestamp, Interval and Binary typed literal are supported.
+     
+      X: Binary Literal
```
Comment (maropu) [inline docs/sql-ref-literals.md:50]: Ah, wait.  The `Binary Literal` section already has examples for this case though, why do we need this new section? In my comment https://github.com/apache/spark/pull/30421#discussion_r530024114, I just asked if `a type constructed value` is a common word in the doc.

Comment (AngersZhuuuu) [reply docs/sql-ref-literals.md:50]: > Ah, wait. The `Binary Literal` section already has examples for this case though, why do we need this new section? In my comment [#30421 (comment)](https://github.com/apache/spark/pull/30421#discussion_r530024114), I just asked if `a type constructed value` is a common word in the doc.

That is what I mentioned in  https://github.com/apache/spark/pull/30496#issuecomment-733536446.

We need to introduce typed literal , this special  grammar. right?

**Comment by AngersZhuuuu** [comment]
Any more update?

**Pull request closed without merging**